QQE-1584.md

```diff
@@ -0,0 +1,82 @@
+(they are not run automatically, as far as I can tell)
+The tests use jlama, which requires Java 21+
+Attempt to run the test locally with jlama fails on my laptop(32.0 GiB RAM, 13th Gen Intel® Core™ i7-1365U × 12). Will try to run it on server, but I am not very optimistic about that.
+Successfully ran it on the server. The downside is there is no rest api, and we have to connect through websocket (which is not optimal). Also, it hangs regularly, even on xlarge.
+
+### Quakriverse extension:
+There are integration tests modules in the extension, but they contain no tests at the moment.
+According to Mario, this is intentional, since they consider upstream lang4chain tests to be enough (but we do not!)
+Ollama modules can be started on the laptop in dev mode. Jlama ones can be run as well.
+
+## Automated test development
+### Functional testing
+1. Add some tests for existing functionality into https://github.com/quarkiverse/quarkus-langchain4j/tree/main/integration-tests/ollama, https://github.com/quarkiverse/quarkus-langchain4j/tree/main/integration-tests/simple-ollama and https://github.com/quarkiverse/quarkus-langchain4j/tree/main/integration-tests/jlama
+2. Beware, that these modules use models of different size, and this can affect reproducibility of tests. Some tests may not work on our hardware with some of the models.
+3. Investigate, whether it is possible to have RAG with ollama/jlama (it should be: [1][2]), add some new functionality and coverage for it into integration tests (maybe in new module)
+4. Add some observability tests.
+5. Add ability to change version of Quarkus and quarkus-langchain4j, which is used in integration tests. Adding new property for version in module pom should be enough.
+6. MR should target main, the reviewer(s) should be Mario and/or Georgios, since they both agreed, that this is the way to go.
+7. Try running the tests on our test servers. If it fails, we may want to try running ollama on a separate machine and connect to it. 
+8. Create a new Jenkins job for prod testing on x86_64 in "general" scope and one for ARM_64 in "extended" scope.
+
+In case our hardware is incapable of running existing models, we should consider using some lighter ones, eg DeepSeek ones or quantizied version of existing LLMs.
+This may lead to more hallucinations in answers, so we should be very careful with out test cases.
+
+### Integration with OpenShiftAI:
+Openshift AI describes itself as "a platform for data scientists and developers of artificial intelligence and machine learning (AI/ML) applications"[3] and mostly focuses on users, who train their own models.
+While some press-releases say[4], that Red Hat offers "third party validated models, available on Hugging Face" all examples I was able to find[5] are Python and Jupyter centric.
+
+Since focus of langchain4j is to help users to work with already trained models, I consider OpenShiftAI to be out-of-scope for this effort.
```

Comment:
What's the PM requirements here?

Reply:
Honestly, considering we have nothing from PM, I'd drop the OpenShift AI note altogether. I know it is frustrating, but there's a lot of notes in here and they're not all that helpful in getting me to understand even the basic mechanics of how this will be supported and delivered, not to mention even further cross-product integrations.

But maybe you have more info as the owner of the effort :)